grails-app/services/org/pih/warehouse/product/ProductMergeService.groovy

```diff
@@ -77,6 +78,9 @@ class ProductMergeService {
          *
          * */
 
+        Product primary = Product.get(primaryId)
+        Product obsolete = Product.get(obsoleteId)
+
```

Comment:
Do we need to validate the existence of these products again? What happens if someone deletes between the validation and merge process?

Also can we add a tech debt ticket to refactor the mergeProduct method to separate each operation into a separate method.

```
def mergeProduct(String primaryId, String obsoleteId) {
    mergeProductSuppliers(primary, obsolete)
    mergeProductComponents(...)
    ... 
    mergeTransaction() 
}
```

We should be able to independently test each of these merge operations.

Reply:
done